pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/apps.kruise.io/v1beta1/StatefulSet/testdata/aggregatestatus-test.yaml

```diff
@@ -86,3 +89,187 @@ statusItems:
+  aggregatedStatus:
+    apiVersion: apps.kruise.io/v1beta1
+    kind: StatefulSet
+    metadata:
+      name: sample
+      namespace: test-statefulset
+      generation: 1
+    spec:   
+      replicas: 2
+      serviceName: sample-statefulset-headless-service
+      podManagementPolicy: Parallel
+      selector:
+        matchLabels:
+          app: sample
+      template:
+        metadata:
+          labels:
+            app: sample
+        spec:
+          volumes:
+            - name: configmap
+              configMap:
+                name: my-sample-config
+          readinessGates:
+            - conditionType: InPlaceUpdateReady
+          containers:
+            - name: nginx
+              image: nginx:alpine
+              ports:
+                - containerPort: 80
+                  name: web
+              env:
+                - name: logData
+                  valueFrom:
+                    configMapKeyRef:
+                      name: mysql-config
+                      key: log
+                - name: lowerData
+                  valueFrom:
+                    configMapKeyRef:
+                      name: mysql-config
+                      key: lower
+      updateStrategy:
+        type: RollingUpdate
+        rollingUpdate:
+          podUpdatePolicy: InPlaceIfPossible
+          maxUnavailable: 2
+    status:
+      replicas: 4
+      readyReplicas: 4
+      currentReplicas: 4
+      updatedReplicas: 4
+      availableReplicas: 4
+      updatedReadyReplicas: 4
+      updateRevision: sample-5675547df7
+      currentRevision: sample-5675547df7
+      observedGeneration: 1
+---
+name: "StatefulSet with partially ready replicas"
+description: "AggregateStatus when some replicas are not ready"
+desiredObj:
+  apiVersion: apps.kruise.io/v1beta1
+  kind: StatefulSet
+  metadata:
+    name: sample
+    namespace: test-statefulset
+    generation: 2
+statusItems:
+  - clusterName: member1
+    status:
+      replicas: 2
+      readyReplicas: 1
+      availableReplicas: 1
+      updatedReplicas: 2
+      updatedReadyReplicas: 1
+      currentReplicas: 2
+      generation: 2
+      observedGeneration: 2
+      resourceTemplateGeneration: 2
+  - clusterName: member2
+    status:
+      replicas: 2
+      readyReplicas: 2
+      availableReplicas: 2
+      updatedReplicas: 2
+      updatedReadyReplicas: 2
+      currentReplicas: 2
+      generation: 2
+      observedGeneration: 2
+      resourceTemplateGeneration: 2
+operation: AggregateStatus
+output:
+  aggregatedStatus:
+    apiVersion: apps.kruise.io/v1beta1
+    kind: StatefulSet
+    metadata:
+      name: sample
+      namespace: test-statefulset
+      generation: 2   
+    status:
+      replicas: 4
+      readyReplicas: 3
+      currentReplicas: 4
+      updatedReplicas: 4
+      availableReplicas: 3
+      updatedReadyReplicas: 3
+      currentRevision: ""
+      updateRevision: ""
+      observedGeneration: 2
+---
+name: "StatefulSet with mixed observed generations"
```

Comment:
The logic regarding generation is actually quite complex. The current use case tests the scenario where observedGeneration and generation are inconsistent on a cluster. Could we add some more test cases to also cover the judgment logic related to resourceTemplateGeneration?